src/systems/trajectory_follower/TrajectoryFollower.hh

```diff
@@ -0,0 +1,152 @@
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace systems
+{
+  // Forward declaration
+  class TrajectoryFollowerPrivate;
+
+  /// \brief A plugin that scripts the movement of a model based on waypoints.
+  /// Note that at this point, the trajectory is always in 2D (x, y).
+  /// The plugin applies a torque until the model is aligned with the next
+  /// waypoint. Then, it applies a force until the model is close enough to
+  /// the waypoint.
+  ///
+  /// This plugin loads a set of waypoints via SDF and traverse the waypoints
+  /// in sequence. The movement is generated applying force and torque to one of
+  /// of the model links. It's also possible to loop through the waypoints for
+  /// generating a never ending trajectory.
+  /// Waypoints may be inserted manually via the <waypoints> element, or
+  /// generated relative to the model's initial position via the <line> or
+  /// <circle> elements.  Only one of these options should be used.
+  ///
+  /// This plugin requires the following SDF parameters:
+  /// * Required parameters:
+  /// <link_name>: The name of the link within the model where the force/torque
+  ///              will be applied when moving the vehicle.
+  ///
+  /// * Optional parameters:
+  /// <loop_forever>: When true, all waypoints will be visited continously in a
+  ///                 circular pattern. If false, the model will stop when the
+  ///                 last waypoint is reached. Note, that if the vehicle moves,
+  ///                 it will still try to reach the very last waypoint.
+  /// <waypoints>: Element specifying the set of waypoints that the
+  ///              the model should navigate through. This block should contain
+  ///              at least one of these blocks:
+  ///                <waypoint>: This block should contain the X, Y of a
+  ///                            waypoint.
+  /// <range_tolerance>: The current waypoint is considered reached if the
+  ///                    distance to it is within +- this tolerance (m).
+  ///                    The default value is 0.5m.
+  /// <bearing_tolerance>: If the bearing to the current waypoint is within
+  ///                      +- this tolerance, a torque won't be applied (degree)
+  ///                      The default value is 2 deg.
+  /// <force>: The force to apply at every plugin iteration in the X direction
+  ///          of the link (N). The default value is 60.
+  /// <torque>: The torque to apply at every plugin iteration in the Yaw
+  ///           direction of the link (Nm). The default value is 50.
+  /// <line>: Element that indicates the model should travel in "line" mode.
+  ///         The block should contain the relative direction and distance from
+  ///         the initial position in which the vehicle should move, specified
+  ///         in the world frame.
+  ///           <direction>: Relative direction (degrees) in the world frame for
```

Comment:
I think most of our angles are specified in radians? For the `<pose>` element in particular, there is a bool [degrees](http://sdformat.org/spec?ver=1.9&elem=model#pose_degrees) attribute to indicate if it's degrees or not. Don't have strong opinion on this though.

Reply:
Changed to radians in [f082490](https://github.com/ignitionrobotics/ign-gazebo/pull/1332/commits/f082490540e4e27aa58065479729b5605044bec7).

Reply:
thanks, I updated doc in [0927042](https://github.com/ignitionrobotics/ign-gazebo/pull/1332/commits/09270428724b15dab8c568aa44d87434961264b4)